### PR TITLE
digital: packed<->unpacked: check chunk sizes, incl. NDBEBUG builds (backport to maint-3.9)

### PR DESCRIPTION
**Pull request by willcode**
Also, get rid of impossible asserts(), and throw exceptions instead of
assert(0)-aborting, as we now have facilities to handle failed
block::work().

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>
(cherry picked from commit c5a1be2e93186df3c20ed2c0512dfd2e1ce7799e)
Signed-off-by: Jeff Long <willcode4@gmail.com>

Backport https://github.com/gnuradio/gnuradio/pull/4255

### CODE REVIEW

**Comment by willcode** [comment]
@marcusmueller Another try to backport failed with lots of link errors on CI. Any idea what's going on?

**Comment by mbr0wn** [comment]
`static constexpr` class members always cause some problems. I can never remember how to properly use them. Maybe you can make a `constexpr` class method `get_bits_per_type()` instead?

**Comment by willcode** [comment]
Thanks for the clue ... `static constexpr` had some changes in C++17.

File: gr-blocks/lib/packed_to_unpacked_impl.h [hunk 1]
```diff
@@ -24,7 +24,7 @@ class packed_to_unpacked_impl : public packed_to_unpacked<T>
     const unsigned int d_bits_per_chunk;
     const endianness_t d_endianness;
     unsigned int d_index;
-    static constexpr unsigned int d_bits_per_type = sizeof(T) * 8;
+    static constexpr unsigned int d_bits_per_type() { return (sizeof(T) * 8); }
```
Comment (mbr0wn) [inline gr-blocks/lib/packed_to_unpacked_impl.h:27]:
```suggestion
    static constexpr unsigned int get_bits_per_type() { return (sizeof(T) * 8); }
```

The `d_` is to indicate non-static class members.

**Review by mbr0wn — CHANGES_REQUESTED** [review]

**Review by mbr0wn — APPROVED** [review]

**Pull request merged**